src/lib.rs

```diff
@@ -59,29 +59,47 @@ pub struct KernelLog {
 }
 
 impl KernelLog {
+
+    const DEFAULT_DEVICE: &'static str = "/dev/kmsg";
+
     /// Create new kernel logger
     pub fn new() -> io::Result<KernelLog> {
         KernelLog::with_level(LevelFilter::Trace)
     }
 
-    /// Create new kernel logger from `KERNLOG_LEVEL` environment variable
+    /// Create new kernel logger from default device with log level specificed by `KERNLOG_LEVEL` environment variable
     pub fn from_env() -> io::Result<KernelLog> {
-        match env::var("KERNLOG_LEVEL") {
-            Err(_) => KernelLog::new(),
-            Ok(s) => match s.parse() {
-                Ok(filter) => KernelLog::with_level(filter),
-                Err(_) => KernelLog::new(),
-            }
-        }
+        Self::from_env_with_device(Self::DEFAULT_DEVICE)
     }
 
-    /// Create new kernel logger with error level filter
+    /// Create new kernel logger from default device with error level filter
     pub fn with_level(filter: LevelFilter) -> io::Result<KernelLog> {
+        Self::with_device_and_level(Self::DEFAULT_DEVICE, filter)
+    }
+
+    /// Create new kernel logger from specific device
+    pub fn with_device(device: &str) -> io::Result<KernelLog> {
+        Self::with_device_and_level(device, LevelFilter::Trace)
+    }
+
+    /// Create new kernel logger from specific device with error level filter
+    pub fn with_device_and_level(device: &str, filter: LevelFilter) -> io::Result<KernelLog> {
```

Comment:
Shouldn't the device type by `Path` or `impl AsRef<Path>`?

Reply:
maybe `impl AsRef<Path>` is better, as the std lib always use this.